presto-docs/src/main/sphinx/security/ldap.rst

```diff
@@ -0,0 +1,239 @@
+===================
+LDAP Authentication
+===================
+
+Presto can be configured to enable frontend LDAP authentication over
+HTTPS for clients, such as the :ref:`cli_ldap`, or the JDBC and ODBC
+drivers. At present only simple LDAP authentication mechanism involving
+user name and password is supported. The Presto client sends a user name 
+and password to the coordinator and coordinator validates these
+credentials using an external LDAP service.
+
+To enable LDAP authentication for Presto, configuration changes are made on
+the Presto coordinator. No changes are required to the worker configuration;
+only the communication from the clients to the coordinator is authenticated.
+
+Presto Server Configuration
+---------------------------
+
+Environment Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. _ldap_server:
+
+LDAP Server
+~~~~~~~~~~~
+
+You can configure the Presto coordinator to authenticate against one
+of the following types of LDAP server implementations:
+
+    * Active Directory
+    * OpenLDAP
+
+Enable TLS
+**********
+
+We recommend to always use Secure LDAP (LDAPS) in production. So before 
+configuring Presto for LDAP authentication, make sure that you have
+TLS enabled on your LDAP server. You need to import the LDAP server's
+TLS certificate to the default Java truststore of the Presto coordinator
+to secure TLS connection. You can use the `keytool` command to
+import this certificate ``ldap_server.crt`` to the truststore
+on the coordinator.
+
+.. code-block:: none
+
+    $ keytool -import -keystore <JAVA_HOME>/jre/lib/security/cacerts -trustcacerts -alias ldap_server -file ldap_server.crt
+
+Java Keystore File for TLS
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using LDAP authentication, access to the Presto coordinator should be
+through HTTPS. You can do it by creating a :ref:`server_java_keystore` on
+the coordinator.
+
+Presto Coordinator Node Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You must make the following changes to the environment prior to configuring the
+Presto coordinator to use LDAP authentication and HTTPS. 
+
+ * :ref:`ldap_server`
+ * :ref:`server_java_keystore`
```

Comment:
I believe you can move this entire section titled `Presto Coordinator Node Configuration` above, instead of back referencing it. The text starting with "You need to import the LDAP server's TLS certificate to the default Java truststore of the Presto coordinator" can be put under a subsection called `TLS Configuration on Presto Coordinator`.